extensions/amp-ad/0.1/utils.js

```diff
@@ -0,0 +1,42 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/** @const {!Object} */
+export const ValidAdContainerTypes = {
+  'AMP-STICKY-AD': 'sa',
+  'AMP-FX-FLYING-CARPET': 'fc',
+  'AMP-LIGHTBOX': 'lb',
+};
+
+/**
+ * Returns a map containing all ad containers that enclose the given ad
+ * element. The params of the map are restricted to the ValidAdContainerTypes,
+ * and the values are all true.
+ * @param {!Element} adElement
+ * @return {!Object<string, boolean>}
+ */
+export function getAllParentAdContainers(adElement) {
```

Comment:
Agreed that this doesn't get *all* parent ad containers. I will change the name to reflect that. But, as I mentioned in a previous comment, this function is more general than simply getting the containers that might prohibit multi-size. It's also used to construct the 'act' URL parameter, for example.